tez-runtime-library/src/main/java/org/apache/tez/runtime/library/common/shuffle/Fetcher.java

```diff
@@ -563,7 +564,7 @@ private HostFetchResult setupConnection(Collection<InputAttemptIdentifier> attem
 
   protected void setupConnectionInternal(String host, Collection<InputAttemptIdentifier> attempts)
       throws IOException, InterruptedException {
-    input = httpConnection.getInputStream();
+    input = new MeasuredDataInputStream(httpConnection.getInputStream());
```

Comment:
-1 for making it the default. Note that System.currentTimeMillis may turn out to be expensive when number of calls are higher. Especially in cloud nodes where hypervisors are involved.

Would suggest to make it optional.

Reply:
thanks for the input @rbalamohan, yes, absolutely, this is a work-in-progress patch, I was thinking about making it configurable too...working on that
I hope the basic idea makes sense to you, counting the time spent in InputStream.read to reflect the actual time spent with network traffic

Reply:
I didn't realize that it was WIP patch. sure, +1 on the idea.